test/test_ssa_direct_qmc.cc

```diff
@@ -28,14 +29,33 @@ namespace KHineq {
     constexpr double a_b  = (base%2)?((base-1.)/4.):((base*base)/(4.*(base+1.)));
     constexpr double f_b  = a_b/std::log(base);
     constexpr double c_b  = (2.0>1.0+1.0/base+a_b)?(2.0):(1.0+1.0/base+a_b);
+    
+    template <typename Fun>
+    void kh_test(rdmini::ssa_direct<size_t,double> ssa, const Fun& f, double V_f, double exact_mu, std::size_t n_events) {
+        rdmini::counting_generator Rlin;
+        rdmini::vdc_uniform_real_distribution<double> U_vdc(0.,1.);
+        double approx_mu =0.;
+        std::ostringstream os;
+        for (std::size_t N=0; N<n_events; ++N) {
+            approx_mu = ( approx_mu*N + f(ssa.inverse_cdf(U_vdc(Rlin))) )/(N+1);
+            if (std::abs(approx_mu-exact_mu)>V_f*(f_b*std::log(N+1)/(N+1)+c_b/(N+1))) {
+                os << "After " << N << " iterations:"
+                    << " expected max error " << f_b*std::log(N+1)/(N+1)+c_b/(N+1)
```

Comment:
This expression appears twice; better to calculate the value once and use it both here and in the test on line 41. (Also, this copy seems to be missing the V_f` factor.)